### PR TITLE
airbnb config extended and overrode

### PR DESCRIPTION
**Pull request by arturwita**
Changes allowing our npm package to work implemented.

### CODE REVIEW

**Comment by arturwita** [comment]
check carefully package.json if all names are properly changed

File: package.json [hunk 1]
```diff
@@ -1,22 +1,31 @@
 {
-  "name": "@squadcraft/eslint-config",
+  "name": "@squadcraft/eslint-config-squadcraft",
   "version": "1.0.0",
```
Comment (Batuzz) [inline package.json:3]: 1.0.0-rc if u want to publish it and check only, 1.0.0 if we will publish it after all without testing :)

Comment (Batuzz) [reply package.json:3]: `rc` stands for `release candidate`

File: package.json [hunk 2]
```diff
@@ -1,22 +1,31 @@
...
+    "eslint-config",
+    "eslint-config-squadcraft",
+    "squadcraft"
   ],
   "author": "arturwita",
   "license": "ISC",
   "bugs": {
-    "url": "https://github.com/arturwita/eslint-config/issues"
+    "url": "https://github.com/arturwita/eslint-config-squadcraft/issues"
   },
```
Comment (Batuzz) [inline package.json]: this can be totally removed

File: package.json [hunk 3]
```diff
@@ -1,22 +1,31 @@
...
-  "homepage": "https://github.com/arturwita/eslint-config#readme"
+  "homepage": "https://github.com/arturwita/eslint-config-squadcraft#readme",
```
Comment (Batuzz) [inline package.json]: same here I guess

File: package.json [hunk 4]
```diff
@@ -1,22 +1,31 @@
...
+  "devDependencies": {
+    "eslint": "^6.8.0",
+    "eslint-config-airbnb-base": "^14.1.0",
+    "eslint-plugin-import": "^2.20.1",
+    "eslint-plugin-node": "^11.0.0"
+  }
```
Comment (Batuzz) [inline package.json]: this time I'm pretty sure those are dependencies, but please check if it can be dev-deps :)

Comment (arturwita) [reply package.json]: check out airbnb's config-base [here](https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb-base/package.json). They have both eslint and eslint plugins stored in dev-deps.

File: package.json (hunk 2 above)
Comment (arturwita) [reply package.json]: removed

File: package.json (hunk 3 above)
Comment (arturwita) [reply package.json]: removed

File: package.json (hunk 1 above)
Comment (arturwita) [reply package.json:3]: How are we going to test our package. Is that even possible? The only thing that comes to my mind is checking:
- eslint's and airbnb's packages deprecation and their compability
- rules deprecation

There is not any coded functionality that needs to be tested. 
Imo we can leave version 1.0.0 as it is, and publish our package when it's ready to use :)

File: package.json (hunk 4 above)
Comment (Batuzz) [reply package.json]: yes, but then - will the package work in main repo, where main repo doesn't have eslint installed separately? check it out

File: package.json (hunk 1 above)
Comment (Batuzz) [reply package.json:3]: to make eslint and airbnb packages unchanged just remove `^` in package.json next to them.
Also - version can be 1.0.0-rc to test it in main repo - that's how you test packages.
You can also use npm link but it depends how you like it.

File: index.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+      RULE_ON,
+      4,
+      {
+       'SwitchCase': RULE_WARN_ONLY
+      }
+    ],
+    'max-classes-per-file': RULE_OFF,
+    'no-restricted-syntax': RULE_OFF,
+    'prefer-object-spread': RULE_ON,
+    'strict': ['error', 'global'],
```
Comment (Batuzz) [inline index.js]:
```suggestion
    'strict': [RULE_ON, 'global'],
```

**Review by Batuzz — APPROVED** [review]

**Pull request merged**